tests/components/homewizard/test_init.py

```diff
@@ -93,3 +94,36 @@ async def test_load_removes_reauth_flow(
     # Flow should be removed
     flows = hass.config_entries.flow.async_progress_by_handler(DOMAIN)
     assert len(flows) == 0
+
+
+@pytest.mark.usefixtures("mock_homewizardenergy")
+async def test_disablederror_reloads_integration(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_homewizardenergy: MagicMock,
+) -> None:
+    """Test DisabledError reloads integration."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    # Make sure current state is loaded and not reauth flow is active
+    assert mock_config_entry.state is ConfigEntryState.LOADED
+    flows = hass.config_entries.flow.async_progress_by_handler(DOMAIN)
+    assert len(flows) == 0
+
+    # Simulate DisabledError and reload to trigger reload
+    mock_homewizardenergy.device.side_effect = DisabledError()
+
+    with pytest.raises(UpdateFailed):
+        await mock_config_entry.runtime_data._async_update_data()
```

Comment:
This is testing the internals, and should be avoided.
You should instead trigger the data update through normal coordinator interval trigger.

Reply:
Changed to use freezegun. It this what you are looking for?

Reply:
Yes thanks 👍